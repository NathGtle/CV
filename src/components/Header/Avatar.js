import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

const query = graphql`
  query {
    placeholderImage: file(relativePath: { eq: "avatar.jpg" }) {
      childImageSharp {
        fluid(maxWidth: 300) {
          ...GatsbyImageSharpFluid
        }
      }
    }
  }
`

export default () => (
  <StaticQuery
    query={query}
    render={data => <Img className="photo" fluid={data.placeholderImage.childImageSharp.fluid} />}
  />
)
