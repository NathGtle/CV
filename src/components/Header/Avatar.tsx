import React from 'react'
import PropTypes from 'prop-types'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

const Avatar = ({ alt }: AvatarInterface) => {
  const data = useStaticQuery(graphql`
    query {
      placeholderImage: file(relativePath: { eq: "avatar.jpg" }) {
        childImageSharp {
          fluid(maxWidth: 300) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)
  return <Img alt={alt} className="photo" fluid={data.placeholderImage.childImageSharp.fluid} />
}

Avatar.defaultProps = {
  alt: 'Kevin MARTIN',
}

Avatar.propTypes = {
  alt: PropTypes.string.isRequired,
}

export default Avatar
