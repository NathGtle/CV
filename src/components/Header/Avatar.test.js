import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'

import Avatar from './Avatar'

jest.mock('gatsby-image', () => 'mock-img')

beforeEach(() => {
  useStaticQuery.mockImplementationOnce(() => ({
    placeholderImage: {
      childImageSharp: {
        fluid: 'image',
      },
    },
  }))
})

describe('Avatar component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Avatar alt="alt" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
