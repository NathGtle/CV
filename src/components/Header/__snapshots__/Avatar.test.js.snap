// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders correctly 1`] = `
<mock-img
  alt="alt"
  className="photo"
  fluid="image"
/>
`;
